File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0
        right = len(height) - 1
        max_unit = 0
        while left < right:
            min_height = min(height[left], height[right])
            curr_unit = (min_height) * (right - left)
            max_unit = max(max_unit, curr_unit)
            if height[left] < height[right]:
                left += 1
            else:
                right -= 1
        return max_unit

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0;
        int right = height.length - 1;
        int maxVal = 0;
        while(left < right){
            int width = right - left;
            int len = Math.min(height[left], height[right]);
            int res = width * len;
            maxVal = Math.max(maxVal, res);
            if(height[left] < height[right]){
                left++;
            } else {
                right--;
            }
        }
        return maxVal;
    }
}